R/income_tax.R

```diff
@@ -31,18 +32,18 @@
 #' @export income_tax
 
 income_tax <- function(income,
-                       fy.year,
+                       fy.year = NULL,
                        age = NULL,
                        family_status = "individual",
                        n_dependants = 0L,
                        .dots.ATO = NULL,
                        return.mode = c("numeric", "integer"),
                        allow.forecasts = FALSE,
                        .debug = FALSE) {
-  if (missing(fy.year)){
-    stop("fy.year is missing, with no default")
+  if (is.null(fy.year)){
+    fy.year <- fy.year(format(as.Date(Sys.Date()),"%Y"))
```

Comment:
OK now to matters of substance:

1. `Sys.Date()` is already a date, so you don't need `as.Date()`. 

2. More importantly, is this function future-proof? What does `fy.year(format(as.Date("2018-07-02"), "%Y"))` return?

3. Please add a test that demonstrates not just the existence of a warning, but also that the value returned is expected. To do this you'll need to surround `income_tax(.)` with `suppressWarnings` so you don't get a spurious message about warnings in your tests.

Reply:
For point 2, do you think it's best to edit the function fy.year or handle it inside income_tax?

Reply:
Just realised its worth doing it in fy.year as the function gets called in other places

Reply:
So when `Sys.Date` is 2018-07-02 (i.e. the second of July this year), the financial year should return 2018-19 not 2017-18 as it currently does (since 2018-07-02 is in the next financial year). 

The function `fy.year` is correct as described, though I haven't used it much recently, preferring `yr2fy` as it's a bit more declarative. It takes a year *ending* and returns the corresponding financial year. 

The function you want is `date2fy`